src/diffusers/utils/testing_utils.py

```diff
@@ -1,10 +1,12 @@
+import inspect
 import os
 import random
 import re
 import unittest
+import warnings
 from distutils.util import strtobool
 from pathlib import Path
-from typing import Union
+from typing import Any, Dict, Optional, Union
 
 import torch
```

Comment:
I think this breaks `import diffusers` if PyTorch is not installed.

We should move `deprecate` to its own file. Not sure whether we should call it `deprecate.py`, `deprecation_utils.py` or something more general in case we need more functionality in the future.

Reply:
Sounds good! I'll add it to `deprecation_utils.py` 

Regarding `import torch`, this should be done in another PR - this is unrelated to this PR (`import torch` was there before). We should definitely make sure though that the library doesn't break when `torch` is not installed.

Reply:
> Regarding `import torch`, this should be done in another PR - this is unrelated to this PR (`import torch` was there before).

Yes, but `testing_utils` was only imported when testing, that's really the difference (not the `import`). I should have made it clearer in my comment :)